code/modules/ruins/spaceruin_code/hilbertshotel.dm

```diff
@@ -46,12 +46,15 @@ GLOBAL_VAR_INIT(hhmysteryRoomNumber, 1337)
     var/chosenRoomNumber = input(user, "What number room will you be checking into?", "Room Number") as null|num
```

Comment:
Take the var as text, check if all chars are algorisms, maybe check for a max length.
Forget about the whole number thing, drop the cap.